core/DataTable/Row.php

```diff
@@ -73,60 +66,33 @@ class Row implements \ArrayAccess, \IteratorAggregate
      */
     public function __construct($row = array())
     {
-        $this->c[self::COLUMNS] = array();
-        $this->c[self::METADATA] = array();
-        $this->c[self::DATATABLE_ASSOCIATED] = null;
-
         if (isset($row[self::COLUMNS])) {
-            $this->c[self::COLUMNS] = $row[self::COLUMNS];
+            $this->columns = $row[self::COLUMNS];
         }
         if (isset($row[self::METADATA])) {
-            $this->c[self::METADATA] = $row[self::METADATA];
-        }
-        if (isset($row[self::DATATABLE_ASSOCIATED])
-            && $row[self::DATATABLE_ASSOCIATED] instanceof DataTable
-        ) {
-            $this->setSubtable($row[self::DATATABLE_ASSOCIATED]);
-        }
-    }
-
-    /**
-     * Because $this->c[self::DATATABLE_ASSOCIATED] is negative when the table is in memory,
-     * we must prior to serialize() call, make sure the ID is saved as positive integer
-     *
-     * Only serialize the "c" member
-     * @ignore
-     */
-    public function __sleep()
```

Comment:
Sleep is no longer needed as we do only serialize the array instead of the whole `Row` instance